src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -647,6 +651,29 @@ void Selection_CollectionChanged(object sender, System.Collections.Specialized.N
         {
             dynamoViewModel.CopyCommand.RaiseCanExecuteChanged();
             dynamoViewModel.PasteCommand.RaiseCanExecuteChanged();
+
+            if (manipulatorDaemon == null)
+                return;
+
+            if (e.Action != NotifyCollectionChangedAction.Move)
+            {
+                if (e.OldItems != null)
+                {
+                    foreach (var nm in e.OldItems.OfType<NodeModel>())
+                        manipulatorDaemon.KillManipulators(nm);
+                }
+            }
+
+            if (e.Action == NotifyCollectionChangedAction.Reset)
+            {
+                manipulatorDaemon.KillAll();
+            }
+
+            if (e.NewItems == null)
+                return;
+
+            foreach (var nm in e.NewItems.OfType<NodeModel>())
+                manipulatorDaemon.CreateManipulator(nm, this);
         }
```

Comment:
@Benglin @aparajit-pratap Nice, so this is where manipulators are created and destroyed.  They are "attached" to the node corresponding to different selection changes.  We might want to move this code to a specific method to clarify.